turbo.json

```diff
@@ -1,14 +1,14 @@
 {
   "pipeline": {
     "build": {
-      "dependsOn": ["intl-compile", "codegen"],
+      "dependsOn": ["^build", "intl-compile", "codegen"],
       "outputs": ["dist/**"]
     },
     "codegen": {
-      "outputs": ["src/api/generated.ts", "src/js/api/generated.ts"]
+      "outputs": ["src/api/generated.ts"]
     },
     "dev": {
-      "dependsOn": ["build:css", "intl-compile", "codegen"],
+      "dependsOn": ["^build", "build:css", "intl-compile", "codegen"],
```

Comment:
I think there maybe something wrong with the caching logic.  If I run "refresh all" twice in a row it appears to rebuild (almost) everything rather than relying on cached layers.
![image](https://user-images.githubusercontent.com/8978655/219159672-1f00b347-9203-4a2b-9cf7-8aef8fb78280.png)
Same thing is happening with the tests:
![image](https://user-images.githubusercontent.com/8978655/219168111-410e31cc-c1b6-4713-8288-bcb9f8b10498.png)

Reply:
Seems to be something with the maintenance container. If i just run just `npm run dev` i get the full cache 🤔 

```sh
 Tasks:    23 successful, 23 total
Cached:    23 cached, 23 total
  Time:    247ms >>> FULL TURBO
```

I did some stuff but it didn't seem to make much of a difference 😢 

 - [fix script permissions](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5697/commits/06d1cdd1fe958cf7b3c3421e340094437a46d69a)
 - [fix script permissions](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5697/commits/06d1cdd1fe958cf7b3c3421e340094437a46d69a)
 - [adjust turbo repo outputs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5697/commits/21cd6f49b02caa6d39ebdb66cb1f45cb862f22ae)

Reply:
Unfortunately, this is still happening for me. :cry:  I'm using the build tasks with the maintenance container.

Reply:
Yeah, Matt was having a lot of problems with the maintenance container as well. There is something funky going on in it that is not allowing turbo do its thing. 😢

Reply:
I still have no idea what is going on here. The permissions look fine (it isn't a read issue). What I am seeing is the workspace hash is different on each run which is a big problem because that is[ how turbo determines hits vs misses](https://turbo.build/repo/docs/core-concepts/caching#hitting-the-cache:~:text=Turborepo%20will%20evaluate%20the%20inputs%20to%20your%20task%20(by%20default%20all%20non%2Dgit%2Dignored%20files%20in%20the%20workspace%20folder)%20and%20turn%20them%20into%20a%20hash%20(e.g.%2078awdk123).). Would this be related to the container itself being volatile? 

When I run any maintenance script, the container only lasts long enough to run the script then it destroys itself to be rebuilt from cache on the next run.

There is a moderately related [highlight with issues with docker in the turbo docs](https://turbo.build/repo/docs/handbook/deploying-with-docker#:~:text=TL%3BDR%3A%20In%20a%20monorepo%2C%20unrelated%20changes%20can%20make%20Docker%20do%20unnecessary%20work%20when%20deploying%20your%20app.) but I cannot tell how related it is to the issue we are encountering here

Reply:
Is it possible that it's another artifact of not mounting the entire workspace in to the maintenance container?  Is it saving its state in an ephemeral directory? Can you see what is changing between runs in the container and causing the miss?

Reply:
I'm not seeing any actual changes, I made some changes in:

- [update some dev tasks, add inputs to turbo config](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5697/commits/c3867a27bcc4ff57e36c0ab30338692a5882c457)
- [update inputs to include package.json files](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5697/commits/d6ea42f1058765690026cf56c418cfdc744d460c)

This effectively narrows down the files turbo looks at to build the cache. But, alas, it did nothing again 😢 I think your guess about the ephemeral directory is accurate when it looks to build the cache but I am stumped 🤷‍♀️